### PR TITLE
Add tests for transfer of tags and dm status on room upgrade

### PR DESCRIPTION
**Pull request by anoadragon453**
Synapse PR: https://github.com/matrix-org/synapse/pull/4412

Adds 2 tests to check that the same room tags and `m.direct` status is set on a room before and after an upgrade.

### CODE REVIEW

File: tests/30rooms/60version_upgrade.pl [hunk 1]
```diff
@@ -59,6 +61,45 @@ sub upgrade_room {
...
+
+Check if a room ID is considered direct by the given user.
+
+=cut
+
+sub is_direct_room {
+   my ( $user, $room_id ) = @_;
+
+   # Download account data events from sync
+   matrix_sync( $user )->then( sub {
```
Comment (richvdh) [inline tests/30rooms/60version_upgrade.pl]: How about using `GET /_matrix/client/r0/user/{user}/account_data/m.direct` (https://github.com/matrix-org/matrix-doc/issues/1339) ?

Comment (anoadragon453) [reply tests/30rooms/60version_upgrade.pl]: Ah cool.

Comment (anoadragon453) [reply tests/30rooms/60version_upgrade.pl]: Indeed this is already a function: https://github.com/matrix-org/sytest/blob/665210b5e13e4adf9194512d44677a2a9ef7ed0a/tests/10apidoc/44account_data.pl#L71-L79

File: tests/30rooms/60version_upgrade.pl [hunk 2]
```diff
@@ -59,6 +61,43 @@ sub upgrade_room {
    });
 }
 
+=head2 is_direct_room
+
+    is_direct_room( $user, $room_id )->then( sub {
+        my ( $is_direct ) = @_;
+    })
+
+Check if a room ID is considered direct by the given user.
```
Comment (richvdh) [inline tests/30rooms/60version_upgrade.pl]: I found this slightly hard to parse; how about something like:

```suggestion
Check if a room ID is considered to be a direct chat by the given user.
```

File: tests/30rooms/60version_upgrade.pl [hunk 3]
```diff
@@ -59,6 +61,43 @@ sub upgrade_room {
...
+
+         # Return whether the given room ID is in the response
+         foreach my $room (@$room_ids) {
+            if ( $room eq $room_id ) {
+               return Future->done( 1 );
+            }
+         }
+      }
+
+      # Didn't find an direct romo with our room ID
```
Comment (richvdh) [inline tests/30rooms/60version_upgrade.pl]: "an direct romo"?
